### ISSUE TITLE
IDEA: CodeBuddies Connect (weekly ask)

### ISSUE THREAD

**Issue by lpatmo** [codebuddies/codebuddies#876]
Many awesome people join and introduce themselves in the #introduce-yourself channel, but it can still be hard to dive into the community on Slack, and shy people sometimes hesitate to ask for help. Connection requests and messages frequently get lost because of the ephemeral nature of chat.

### PROPOSAL:
Every week, ask the question "What are you looking for this week?" in a community-wide notification.

Participants can choose from a number of options, and will be presented with follow-up suggestions or questions which they must respond to. 

After they submit the form, they'll be able to see a board of responses organized by option category, and comment on the responses if they'd like to follow up. (Think: Trello board where each card is a response, and can be commented on)

(In addition, we can also experiment with "matching" people who've demonstrated similar interests this week with each other, either manually by connecting people via email or -- later -- via an algorithm.)

## What are you looking for this week?
[ ] Algorithms coding practice w/ a pair programming partner
- What language(s) do you want to use? _____ 
- What timezone are you in? _____
- When are you generally free this week? _____

[ ] An interviewing practice partner 
- What resource are you using? _____
- What timezone are you in? _____
- When are you generally free this week? _____

[ ] Motivation to finish (or continue working on) a tutorial or MOOC
- Please link to the course or tutorial you are working on _____
- Are you interested in jumping into a hangout with others who are working through the same course? (Yes / No)
     - Yes, a silent hangout
          - What timezone are you in? _____
          - When are you generally free this week? _____
     - Yes, a collaborative hangout (discuss out loud)
          - What timezone are you in? _____
          - When are you generally free this week? _____
     - No, but I would like to lurk in a study group created for this course
- Would you be willing to start a study group for this course/tutorial, if it doesn't already exist?
     - Yes
           - Thanks for stepping up! You can create it here: codebuddies.org/study-groups. Please link it here after you've done so: __________
     - No, I'd rather someone else start a group
 
[ ] Feedback on a personal project
- Please link to the personal project you are working on _____
- What technologies does it use? _____
- Is there any particular feedback you are looking for? _____

[ ] Contributors to an open-sourced personal project
- Please link to the open-sourced project that is open to contributors _____
- What technologies does it use? _____
- What do you need help with? _____
- Would you be willing to schedule a hangout to walk new contributors through the codebase? (Yes/No)
- Would you be willing to schedule a hangout to work on the project in real-time, and let participants jump in to help -- or at a minimum, observe? (Yes/No)

[ ] An open-sourced project to contribute to
- What technologies are you interested in contributing to? _____

[ ] Coding help
- What is the problem you're trying to solve? _____
- What have you tried so far? (Please link to a code sample if possible.) _____
- Are you looking for live help in a hangout where you’d be screensharing? (Yes/No)
    - No
        - Please ask your question and share what you've tried in a relevant channel on Slack
    - Yes
        - What timezone are you in? _____
        - When are you generally available this week? _____

[ ] An accountability partner
- What are you looking for in an accountability partner? _____
- Are you looking for silent focus/silent study motivation? (Yes/No)
    - Yes
         - Join our global silent productivity hangout here: codebuddies.org/coworking

[ ] Career advice
- What is your question? _____

[ ] I want to practice teaching something, or practice a talk
- What do you want to teach or give a talk on? _____
- What timezone are you in? _____
- When are you generally available this week? _____
(People who're interested can comment. Then the OP will schedule a `teaching` hangout on the site.)

[ ] I want to help mentor someone on Slack
Note to mentors: (content below drafted by @sergeant-q)
```
1 - Set expectations and goals with your mentee  (how often are you meeting and what does the mentee want to learn?)
2 - Show them _how_ to do something and not do it for them
3 - Encourage them to constantly be learning (google is your best friend)
4 - Hold your mentee accountable for any follow up assignments and showing up! (remember, you are volunteering your time for free!)
5 - Do more listening
6 - Remember. Your role is to _guide_ not necessarily _teach_ (though you will be doing some of that)
```
- What can you help with? _____

[ ] I want to help mentor someone in a 1-hour hangout
- What can you help with? _____
- What timezone are you in? _____
- When are you generally available this week? _____

[ ] I want a mentor this week on Slack
- What do you want help with? _____

[ ] I want a mentor who can meet with me in a 1-hour hangout
- What do you want help with? _____
- What timezone are you in? _____
- When are you generally available this week? _____

## FEEDBACK WELCOME ON:
1) Is the list of options above too many? If so, how would you cut it down? 
2) Do you have suggestions for other options I haven't thought of?

Let's prototype this idea using a Google Form/Spreadsheet this week before building it out onto the site.

**Comment by lpatmo** [comment]
Here is the full list of options, for easier readability: 

## What are you looking for this week?
[ ] Algorithms coding practice w/ a pair programming partner
[ ] An interviewing practice partner
[ ] Motivation to finish (or continue working on) a tutorial or MOOC
[ ] Feedback on a personal project
[ ] Contributors to an open-sourced personal project
[ ] An open-sourced project to contribute to
[ ] Coding help
[ ] An accountability partner
[ ] Career advice
[ ] I want to practice teaching something, or practice a talk
[ ] I want to help mentor someone on Slack
[ ] I want to help mentor someone in a 1-hour hangout
[ ] I want a mentor this week on Slack
[ ] I want a mentor who can meet with me in a 1-hour hangout

Maybe we can cut this down?

Also, maybe for the very first iteration of this experiment, we can skip the interactive follow-up questions and just have a general description box asking people to elaborate on the option they've selected.

**Comment by lpatmo** [comment]
Revamped the questions based on the responses so far (here is a public spreadsheet: https://docs.google.com/spreadsheets/u/4/d/1iVMwyMv9hEVlZWdfXOh4B9XSmDZ_QW5PCF3bQKfFMXA/edit#gid=0)

# What are you looking for this week?
## A coding or accountability partner
### Can you elaborate on what you'd like to work on with a coding partner?
- Algorithms coding practice (pair programming)
- Interview practice
- Motivation to finish (or continue working on) a tutorial or a MOOC
- An accountability partner I can check in with

## Collaboration on an open-sourced project
- I'd like to contribute to an open-sourced project with some collaborators
- I'm looking for contributors to my open-sourced project

## Feedback, mentorship or advice
- I'd like some feedback on a coding project
- I am looking for career advice
- I am looking for a mentor who can meet with me in a 1-hour hangout
- I am looking for a mentor who can give me feedback on Slack

## I want to teach or mentor
- I want to practice teaching something, or practice giving a talk
- I want to mentor someone on Slack
- I can spare one hour this week to mentor someone in a hangout

**Comment by lpatmo** [comment]
I'm going to pretend this is a sorting hat quiz:

![screen shot 2018-06-09 at 12 45 54 am](https://user-images.githubusercontent.com/4512699/41189178-8b885414-6b7e-11e8-875f-57e057016dd0.jpg)

**Comment by lpatmo** [comment]
![screen shot 2018-06-09 at 1 18 58 am](https://user-images.githubusercontent.com/4512699/41189464-33c34af4-6b83-11e8-96c9-bb2776b7d784.jpg)

Design feedback welcome.

**Comment by sergeant-q** [comment]
@lpatmo the design mock-ups look great. I'd love to see what happens next? I'm trying to visualize these "trello" like cards and what comes of those cards/responses? 

If the goal is to match folks, I think we should really flush out this feature. Sounds very promising in my opinion.

**Comment by billglover** [comment]
This is current assigned to @sebbel, should this be moved to @lpatmo now?

**Comment by ideaguy3d** [comment]
How far along is this? Where can I go to check out the code? I'd be willing to spend about 10 hours a week solving this problem.

**Comment by lpatmo** [comment]
@ideaguy3d Thanks for your interest! We've got some code here: https://github.com/codebuddies/cb-connect. Please join our Discord's #cb-connect channel at https://discord.gg/yvtBmEW. 

I'm happy to do a code walkthrough with you sometime next weekend, if you're interested. Feel free to ask any questions on the Discord channel.

**Comment by adachiu** [comment]
@lpatmo Maybe we could make the page more engaging. I was playing with a CB "Connect" logo and also a potential new look for the page. I have some more ideas to mock up for the different states. More to come next week.

![Connect Main Screen v1](https://user-images.githubusercontent.com/13413986/61438267-a10ea180-a971-11e9-9ae5-fdbe41410877.png)

**Comment by ideaguy3d** [comment]
@lpatmo yeah we can spend an hour or so this weekend on it. I would also be more useful if I could create a separate web service to build this because I'd want to do some heavy duty SQL coding.

@adachiu It looks pretty, but the interaction design is far more useful than the visual design with something this complex. I can see the UI view's going into so many branches and sub branches

**Comment by lpatmo** [comment]
@Julius We might explore a relational DB stack with CodeBuddies V3, but
this project is meant to be a quick MVP so we can connect people. It's
missing a key feature (letting people submit more than one entry) but is
close to being done.

@Ada Love the icons, and I like that you've organized the entry points so
that it's 4 instead of 6. Do you have thoughts on the rest of the signup
flow? Have you seen connect.codebuddies.org? It's the current UI.

**Comment by adachiu** [comment]
@lpatmo Yes, I am basing the rest of the flow I'm doing with the current website content. I will post more progress today.

@ideaguy3d I'm working on the ux of the user flow for this.

**Comment by lpatmo** [comment]
@ideaguy3d There won't be a lot of opportunity for SQL for this project,
sorry. :(

If you're still interested, let's figure out a time to pair on Discord!

**Comment by adachiu** [comment]
I updated the layout of the cards so that it is vertical. This would give us space to have a horizontal expansion of the form.

Sketch Cloud Link: https://sketch.cloud/s/PKPgL

![Connect Main Screen 3](https://user-images.githubusercontent.com/13413986/61508031-cb686980-aa1a-11e9-9b50-c11cede9ee95.png)
![Flow - Partner 1](https://user-images.githubusercontent.com/13413986/61508034-cefbf080-aa1a-11e9-9c14-12d36264ecbe.png)
![Flow - Partner 2](https://user-images.githubusercontent.com/13413986/61508036-d15e4a80-aa1a-11e9-9239-7e18e510c897.png)


I'm working on how it would look like for the other expanded form options.

**Comment by distalx** [comment]
@adachiu, it looks fantastic.

**Comment by adachiu** [comment]
(Copy and Paste from my Slack comment so we can save the thought here)

Perhaps we can add the breadcrumbs that I mentioned earlier.  What if we break it down like this:

First form section is *all about the User*
1) Name 2) Intro 3) My skill(s)

Second form section is all about what the *User wants to do*
1) I’m looking for 2) What skill(s) I want to improve

Third form section is *contact information*
1) Email  2) Time Zone

**Comment by lpatmo** [comment]
What the matching dashboard currently looks like:

![image](https://user-images.githubusercontent.com/4512699/61515801-52d0c100-a9b8-11e9-8a98-924e8908617e.png)

![image](https://user-images.githubusercontent.com/4512699/61515824-5d8b5600-a9b8-11e9-8c83-db4b1609594b.png)
